proxy/envoy/discovery.go

```diff
@@ -396,41 +396,44 @@ func (ds *DiscoveryService) ListEndpoints(request *restful.Request, response *re
 		}
 		var err error
 		if out, err = json.MarshalIndent(hosts{Hosts: hostArray}, " ", " "); err != nil {
-			errorResponse(response, http.StatusInternalServerError, err.Error())
+			errorResponse(response, http.StatusInternalServerError, "SDS " + err.Error())
 			return
 		}
 		ds.sdsCache.updateCachedDiscoveryResponse(key, out)
 	}
 	writeResponse(response, out)
 }
 
-func (ds *DiscoveryService) parseRole(request *restful.Request) (proxy.Node, error) {
-	if sc := request.PathParameter(ServiceCluster); sc != ds.Mesh.IstioServiceCluster {
-		return proxy.Node{}, fmt.Errorf("unexpected %s %q", ServiceCluster, sc)
+func (ds *DiscoveryService) parseDiscoveryRequest(request *restful.Request) (string, string, proxy.Node, error) {
+	cluster := request.PathParameter(ServiceCluster)
+	if cluster != ds.Mesh.IstioServiceCluster {
```

Comment:
I will add a comment, but remark this is not my code, the check was already there. I merely changed the code to return all the fields and display proper error message.